In [1]:
# Importing Libraries
import numpy as np 
import pandas as pd
import matplotlib.pyplot as plt
import os
from itertools import chain
from collections import Counter
import operator as op
# Adjustment Options
pd.set_option('display.max_columns', None) # Display All Columns given the dataset has many columns
pd.set_option('display.max_rows',75)


# Exploring .csv kaggle files
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        if filename.split(".")[-1] == "csv":
            print(os.path.join(dirname, filename))


/kaggle/input/mental-health-in-tech-2016/mental-heath-in-tech-2016_20161114.csv


In [2]:
df = pd.read_csv("/kaggle/input/mental-health-in-tech-2016/mental-heath-in-tech-2016_20161114.csv")
df

/usr/local/lib/python3.11/dist-packages/pandas/io/formats/format.py:1458: RuntimeWarning: invalid value encountered in greater
  has_large_values = (abs_vals > 1e6).any()
/usr/local/lib/python3.11/dist-packages/pandas/io/formats/format.py:1459: RuntimeWarning: invalid value encountered in less
  has_small_values = ((abs_vals < 10 ** (-self.digits)) & (abs_vals > 0)).any()
/usr/local/lib/python3.11/dist-packages/pandas/io/formats/format.py:1459: RuntimeWarning: invalid value encountered in greater
  has_small_values = ((abs_vals < 10 ** (-self.digits)) & (abs_vals > 0)).any()


Are you self-employed?  \
0                          0   
1                          0   
2                          0   
3                          1   
4                          0   
...                      ...   
1428                       1   
1429                       1   
1430                       0   
1431                       0   
1432                       0   

     How many employees does your company or organization have?  \
0                                                26-100           
1                                                  6-25           
2                                                  6-25           
3                                                   NaN           
4                                                  6-25           
...                                                 ...           
1428                                                NaN           
1429                                                NaN           
1430                                            100-500           
1431                                            100-500           
1432                                            100-500           

      Is your employer primarily a tech company/organization?  \
0                                                   1.0         
1                                                   1.0         
2                                                   1.0         
3                                                   NaN         
4                                                   0.0         
...                                                 ...         
1428                                                NaN         
1429                                                NaN         
1430                                                1.0         
1431                                                0.0         
1432                                                1.0         

      Is your primary role within your company related to tech/IT?  \
0                                                   NaN              
1                                                   NaN              
2                                                   NaN              
3                                                   NaN              
4                                                   1.0              
...                                                 ...              
1428                                                NaN              
1429                                                NaN              
1430                                                NaN              
1431                                                1.0              
1432                                                NaN              

     Does your employer provide mental health benefits as part of healthcare coverage?  \
0                       Not eligible for coverage / N/A                                  
1                                                    No                                  
2                                                    No                                  
3                                                   NaN                                  
4                                                   Yes                                  
...                                                 ...                                  
1428                                                NaN                                  
1429                                                NaN                                  
1430                                                Yes                                  
1431                                       I don't know                                  
1432                                                Yes                                  

     Do you know the options for mental health care available under your employer-provided coverage?  \
0        

In [3]:
for i, column in enumerate(df.columns):
    print(i, column)

0 Are you self-employed?
1 How many employees does your company or organization have?
2 Is your employer primarily a tech company/organization?
3 Is your primary role within your company related to tech/IT?
4 Does your employer provide mental health benefits as part of healthcare coverage?
5 Do you know the options for mental health care available under your employer-provided coverage?
6 Has your employer ever formally discussed mental health (for example, as part of a wellness campaign or other official communication)?
7 Does your employer offer resources to learn more about mental health concerns and options for seeking help?
8 Is your anonymity protected if you choose to take advantage of mental health or substance abuse treatment resources provided by your employer?
9 If a mental health issue prompted you to request a medical leave from work, asking for that leave would be:
10 Do you think that discussing a mental health disorder with your employer would have negative consequences?

# Data Cleaning
- Dealing with Nan values
- Dealing with ordinal and nominal columns
- Normalizing Data

In [4]:
# a threshold for Nan values
nan_column_threshold = df.shape[0]*.25
nan_row_threshold = df.shape[1]*.25
nan_column_threshold, nan_row_threshold

(358.25, 15.75)

In [5]:
df.isna().sum()

Are you self-employed?                                                                                                                                                                 0
How many employees does your company or organization have?                                                                                                                           287
Is your employer primarily a tech company/organization?                                                                                                                              287
Is your primary role within your company related to tech/IT?                                                                                                                        1170
Does your employer provide mental health benefits as part of healthcare coverage?                                                                                                    287
Do you know the options for mental health care available under your employe

In [6]:
df.loc[:,df.isna().sum() > nan_column_threshold]

/usr/local/lib/python3.11/dist-packages/pandas/io/formats/format.py:1458: RuntimeWarning: invalid value encountered in greater
  has_large_values = (abs_vals > 1e6).any()
/usr/local/lib/python3.11/dist-packages/pandas/io/formats/format.py:1459: RuntimeWarning: invalid value encountered in less
  has_small_values = ((abs_vals < 10 ** (-self.digits)) & (abs_vals > 0)).any()
/usr/local/lib/python3.11/dist-packages/pandas/io/formats/format.py:1459: RuntimeWarning: invalid value encountered in greater
  has_small_values = ((abs_vals < 10 ** (-self.digits)) & (abs_vals > 0)).any()


Is your primary role within your company related to tech/IT?  \
0                                                   NaN              
1                                                   NaN              
2                                                   NaN              
3                                                   NaN              
4                                                   1.0              
...                                                 ...              
1428                                                NaN              
1429                                                NaN              
1430                                                NaN              
1431                                                1.0              
1432                                                NaN              

     Do you know the options for mental health care available under your employer-provided coverage?  \
0                                                   NaN                                                
1                                                   Yes                                                
2                                                   NaN                                                
3                                                   NaN                                                
4                                                   Yes                                                
...                                                 ...                                                
1428                                                NaN                                                
1429                                                NaN                                                
1430                                                Yes                                                
1431                                      I am not sure                                                
1432                                                 No                                                

      Do you have medical coverage (private insurance or state-provided) which includes treatment of  mental health issues?  \
0                                                   NaN                                                                       
1                                                   NaN                                                                       
2                                                   NaN                                                                       
3                                                   1.0                                                                       
4                                                   NaN                                                                       
...                                                 ...                                                                       
1428                                                1.0                                                                       
1429                                                1.0                                                                       
1430                                                NaN                                                                       
1431                                                NaN                                                                       
1432                                                NaN                                                                       

     Do you know local or online resources to seek help for a mental health disorder?  \
0                                                   NaN                                 
1                                                   NaN                                 
2                                                   NaN                                 
3                                  

In [7]:
df.loc[df.isna().sum(axis=1) > nan_row_threshold,:]

/usr/local/lib/python3.11/dist-packages/pandas/io/formats/format.py:1458: RuntimeWarning: invalid value encountered in greater
  has_large_values = (abs_vals > 1e6).any()
/usr/local/lib/python3.11/dist-packages/pandas/io/formats/format.py:1459: RuntimeWarning: invalid value encountered in less
  has_small_values = ((abs_vals < 10 ** (-self.digits)) & (abs_vals > 0)).any()
/usr/local/lib/python3.11/dist-packages/pandas/io/formats/format.py:1459: RuntimeWarning: invalid value encountered in greater
  has_small_values = ((abs_vals < 10 ** (-self.digits)) & (abs_vals > 0)).any()


Are you self-employed?  \
0                          0   
3                          1   
9                          1   
17                         0   
18                         1   
...                      ...   
1420                       1   
1423                       0   
1428                       1   
1429                       1   
1432                       0   

     How many employees does your company or organization have?  \
0                                                26-100           
3                                                   NaN           
9                                                   NaN           
17                                             500-1000           
18                                                  NaN           
...                                                 ...           
1420                                                NaN           
1423                                            100-500           
1428                                                NaN           
1429                                                NaN           
1432                                            100-500           

      Is your employer primarily a tech company/organization?  \
0                                                   1.0         
3                                                   NaN         
9                                                   NaN         
17                                                  1.0         
18                                                  NaN         
...                                                 ...         
1420                                                NaN         
1423                                                1.0         
1428                                                NaN         
1429                                                NaN         
1432                                                1.0         

      Is your primary role within your company related to tech/IT?  \
0                                                   NaN              
3                                                   NaN              
9                                                   NaN              
17                                                  NaN              
18                                                  NaN              
...                                                 ...              
1420                                                NaN              
1423                                                NaN              
1428                                                NaN              
1429                                                NaN              
1432                                                NaN              

     Does your employer provide mental health benefits as part of healthcare coverage?  \
0                       Not eligible for coverage / N/A                                  
3                                                   NaN                                  
9                                                   NaN                                  
17                                                   No                                  
18                                                  NaN                                  
...                                                 ...                                  
1420                                                NaN                                  
1423                                                Yes                                  
1428                                                NaN                                  
1429                                                NaN                                  
1432                                                Yes                                  

     Do you know the options for mental health care available under your employer-provided coverage?  \
0        

In [8]:
df.loc[df.isna().sum(axis=1) > nan_row_threshold,:].isna().sum(axis=1)

0       17
3       19
9       18
17      16
18      28
        ..
1420    21
1423    26
1428    21
1429    30
1432    26
Length: 494, dtype: int64

## Nan Values
- 16 columns have more than 25% of Nan values but this could be to no answer or inapplicable. to be reviewed.
- 494 respondents didnt answer more than 25% of the survey. Theyll be dropped and considered inaccurate responses. 

In [9]:
df_2 = df.loc[df.isna().sum(axis=1) < nan_row_threshold,:]
df_2.shape

(939, 63)

In [10]:
df_2.loc[:,df_2.isna().sum() > nan_column_threshold] # Checking again for column Nans

/usr/local/lib/python3.11/dist-packages/pandas/io/formats/format.py:1458: RuntimeWarning: invalid value encountered in greater
  has_large_values = (abs_vals > 1e6).any()
/usr/local/lib/python3.11/dist-packages/pandas/io/formats/format.py:1459: RuntimeWarning: invalid value encountered in less
  has_small_values = ((abs_vals < 10 ** (-self.digits)) & (abs_vals > 0)).any()
/usr/local/lib/python3.11/dist-packages/pandas/io/formats/format.py:1459: RuntimeWarning: invalid value encountered in greater
  has_small_values = ((abs_vals < 10 ** (-self.digits)) & (abs_vals > 0)).any()


Is your primary role within your company related to tech/IT?  \
1                                                   NaN              
2                                                   NaN              
4                                                   1.0              
5                                                   NaN              
6                                                   NaN              
...                                                 ...              
1425                                                NaN              
1426                                                NaN              
1427                                                NaN              
1430                                                NaN              
1431                                                1.0              

      Do you have medical coverage (private insurance or state-provided) which includes treatment of  mental health issues?  \
1                                                   NaN                                                                       
2                                                   NaN                                                                       
4                                                   NaN                                                                       
5                                                   NaN                                                                       
6                                                   NaN                                                                       
...                                                 ...                                                                       
1425                                                NaN                                                                       
1426                                                NaN                                                                       
1427                                                NaN                                                                       
1430                                                NaN                                                                       
1431                                                NaN                                                                       

     Do you know local or online resources to seek help for a mental health disorder?  \
1                                                   NaN                                 
2                                                   NaN                                 
4                                                   NaN                                 
5                                                   NaN                                 
6                                                   NaN                                 
...                                                 ...                                 
1425                                                NaN                                 
1426                                                NaN                                 
1427                                                NaN                                 
1430                                                NaN                                 
1431                                                NaN                                 

     If you have been diagnosed or treated for a mental health disorder, do you ever reveal this to clients or business contacts?  \
1                                                   NaN                                                                             
2                                                   NaN                                                                             
4                                                   NaN                                                                             
5                                      

In [11]:
df_2.loc[:,df_2.isna().sum() > nan_column_threshold].columns

Index(['Is your primary role within your company related to tech/IT?',
       'Do you have medical coverage (private insurance or state-provided) which includes treatment of  mental health issues?',
       'Do you know local or online resources to seek help for a mental health disorder?',
       'If you have been diagnosed or treated for a mental health disorder, do you ever reveal this to clients or business contacts?',
       'If you have revealed a mental health issue to a client or business contact, do you believe this has impacted you negatively?',
       'If you have been diagnosed or treated for a mental health disorder, do you ever reveal this to coworkers or employees?',
       'If you have revealed a mental health issue to a coworker or employee, do you believe this has impacted you negatively?',
       'Do you believe your productivity is ever affected by a mental health issue?',
       'If yes, what percentage of your work time (time performing primary or secondary job func

## Fill NaN with inapplicable or no asnwer or drop

In [12]:
# No Answer Columns
# any columns with a determinate answer but with Nan will be dropped
drop_columns = ['Is your primary role within your company related to tech/IT?',
                'Do you have medical coverage (private insurance or state-provided) which includes treatment of \xa0mental health issues?',
                'Do you know local or online resources to seek help for a mental health disorder?'
               ]

# Questions that deal with belief or witness will be filled with 'no answer'
no_asnwers_cols = ['Do you believe your productivity is ever affected by a mental health issue?',
                  'Have your observations of how another individual who discussed a mental health disorder made you less likely to reveal a mental health issue yourself in your current workplace?'
                  ]

# Any 'If questions' will be filled with inapplicable 
inapplicable_cols = ['If you have revealed a mental health issue to a client or business contact, do you believe this has impacted you negatively?',
                    'If you have been diagnosed or treated for a mental health disorder, do you ever reveal this to clients or business contacts?',
                    'If you have been diagnosed or treated for a mental health disorder, do you ever reveal this to coworkers or employees?',
                    'If you have revealed a mental health issue to a coworker or employee, do you believe this has impacted you negatively?',
                    'If yes, what percentage of your work time (time performing primary or secondary job functions) is affected by a mental health issue?',
                    'If yes, what condition(s) have you been diagnosed with?',
                    'If maybe, what condition(s) do you believe you have?',
                    'If so, what condition(s) were you diagnosed with?'
                    ]

In [13]:
df_3 = df_2.drop(columns=drop_columns, axis=1)
df_3

Are you self-employed?  \
1                          0   
2                          0   
4                          0   
5                          0   
6                          0   
...                      ...   
1425                       0   
1426                       0   
1427                       0   
1430                       0   
1431                       0   

     How many employees does your company or organization have?  \
1                                                  6-25           
2                                                  6-25           
4                                                  6-25           
5                                        More than 1000           
6                                                26-100           
...                                                 ...           
1425                                            100-500           
1426                                            100-500           
1427                                           500-1000           
1430                                            100-500           
1431                                            100-500           

      Is your employer primarily a tech company/organization?  \
1                                                   1.0         
2                                                   1.0         
4                                                   0.0         
5                                                   1.0         
6                                                   1.0         
...                                                 ...         
1425                                                1.0         
1426                                                1.0         
1427                                                1.0         
1430                                                1.0         
1431                                                0.0         

     Does your employer provide mental health benefits as part of healthcare coverage?  \
1                                                    No                                  
2                                                    No                                  
4                                                   Yes                                  
5                                                   Yes                                  
6                                          I don't know                                  
...                                                 ...                                  
1425                                                Yes                                  
1426                                       I don't know                                  
1427                                                Yes                                  
1430                                                Yes                                  
1431                                       I don't know                                  

     Do you know the options for mental health care available under your employer-provided coverage?  \
1                                                   Yes                                                
2                                                   NaN                                                
4                                                   Yes                                                
5                                         I am not sure                                                
6                                                    No                                                
...                                                 ...                                                
1425                                      I am not sure                                                
1426                                      I am not sure                                                
1427              

In [14]:
df_3.loc[:,no_asnwers_cols] = df_3.loc[:,no_asnwers_cols].fillna("No Answer")

In [15]:
df_3.loc[:,inapplicable_cols] = df_3.loc[:,inapplicable_cols].fillna("Inapplicable")

In [16]:
df_3.loc[:,df_3.isna().sum() > 0]

Do you know the options for mental health care available under your employer-provided coverage?  \
1                                                   Yes                                                
2                                                   NaN                                                
4                                                   Yes                                                
5                                         I am not sure                                                
6                                                    No                                                
...                                                 ...                                                
1425                                      I am not sure                                                
1426                                      I am not sure                                                
1427                                                 No                                                
1430                                                Yes                                                
1431                                      I am not sure                                                

                                        Why or why not?  \
1     It would depend on the health issue. If there ...   
2     They would provable need to know, to Judge if ...   
4     Depending on the interview stage and whether I...   
5     If it would potentially affect my ability to d...   
6     I want to gauge their ability to support this ...   
...                                                 ...   
1425                                                NaN   
1426                                                NaN   
1427   I would be afraid it would reflect poorly on me.   
1430  Fear that doing so would cause the employer to...   
1431                          Stigma with some diseases   

                                      Why or why not?.1  \
1     While mental health has become a more prominen...   
2                                      Stigma, mainly.    
4                                          I don't know   
5     It would depend on the field & what I knew of ...   
6     I want to gauge their ability to support, unde...   
...                                                 ...   
1425                                            Stigma.   
1426                                                NaN   
1427  I would be certain it would reflect poorly on me.   
1430  Fear that the employer would consider addition...   
1431  Feels like I'm making a mountain out of a mole...   

     Have you observed or experienced an unsupportive or badly handled response to a mental health issue in your current or previous workplace?  \
1                                                    No                                                                                           
2                                        Maybe/Not sure                                                                                           
4                                    Yes, I experienced                                                                                           
5                                    Yes, I experienced                                                                                           
6                                                    No                                                                                           
...                                                 ...                                                                                           
1425                                                 No                                                                                           
1426                                 Yes, I experienced                                                                                           
1427           

In [17]:
for column in df_3.loc[:,df_3.isna().sum() > 0].columns:
    print(f"{column}:\t{df_3[column].isna().sum()}")

Do you know the options for mental health care available under your employer-provided coverage?:	85
Why or why not?:	189
Why or why not?.1:	164
Have you observed or experienced an unsupportive or badly handled response to a mental health issue in your current or previous workplace?:	28
What is your gender?:	3
What US state or territory do you live in?:	306
What US state or territory do you work in?:	302


In [18]:
df_4 = df_3.copy()

In [19]:
df_4.loc[:,df_4.isna().sum() > 0] = df_4.loc[:,df_4.isna().sum() > 0].fillna("No Answer")

In [20]:
df_4.loc[:,df_4.isna().sum() > 0]

Empty DataFrame
Columns: []
Index: [1, 2, 4, 5, 6, 7, 8, 10, 11, 12, 13, 14, 15, 16, 19, 20, 21, 23, 26, 27, 28, 29, 31, 34, 35, 36, 37, 38, 39, 41, 42, 44, 45, 47, 49, 50, 51, 53, 54, 55, 57, 58, 59, 60, 61, 62, 66, 67, 68, 69, 72, 73, 74, 77, 78, 79, 80, 81, 85, 86, 87, 90, 91, 92, 93, 94, 95, 96, 97, 99, 100, 102, 103, 106, 107, 110, 111, 115, 116, 117, 118, 119, 120, 121, 123, 126, 128, 129, 130, 131, 132, 133, 134, 135, 136, 137, 138, 139, 140, 141, ...]

[939 rows x 0 columns]

All Nan Values have been adjusted

In [21]:
df_4.reset_index(drop=True,inplace=True)

## Noramlizing Answers in the Dataset

In [22]:
df_4.head()

Are you self-employed?  \
0                       0   
1                       0   
2                       0   
3                       0   
4                       0   

  How many employees does your company or organization have?  \
0                                               6-25           
1                                               6-25           
2                                               6-25           
3                                     More than 1000           
4                                             26-100           

   Is your employer primarily a tech company/organization?  \
0                                                1.0         
1                                                1.0         
2                                                0.0         
3                                                1.0         
4                                                1.0         

  Does your employer provide mental health benefits as part of healthcare coverage?  \
0                                                 No                                  
1                                                 No                                  
2                                                Yes                                  
3                                                Yes                                  
4                                       I don't know                                  

  Do you know the options for mental health care available under your employer-provided coverage?  \
0                                                Yes                                                
1                                          No Answer                                                
2                                                Yes                                                
3                                      I am not sure                                                
4                                                 No                                                

  Has your employer ever formally discussed mental health (for example, as part of a wellness campaign or other official communication)?  \
0                                                Yes                                                                                       
1                                                 No                                                                                       
2                                                 No                                                                                       
3                                                 No                                                                                       
4                                                 No                                                                                       

  Does your employer offer resources to learn more about mental health concerns and options for seeking help?  \
0                                                Yes                                                            
1                                                 No                                                            
2                                                 No                                                            
3                                                Yes                                                            
4                                                 No                                                            

  Is your anonymity protected if you choose to take advantage of mental health or substance abuse treatment resources provided by your employer?  \
0                                                Yes                                                                                               
1                                       I don't know                                                                        

In [23]:
one_unique_answer = []
for i,column in enumerate(df_4.columns):
    if len(df_4[column].unique()) > 1:
        print(f"****{i, column}****")
        print(df_4[column].unique())
        print("\n")
    else:
        one_unique_answer.append(column)
print("The following columns have one unique answer:")
print(*one_unique_answer,sep="\n")
    

****(1, 'How many employees does your company or organization have?')****
['6-25' 'More than 1000' '26-100' '100-500' '1-5' '500-1000']


****(2, 'Is your employer primarily a tech company/organization?')****
[1. 0.]


****(3, 'Does your employer provide mental health benefits as part of healthcare coverage?')****
['No' 'Yes' "I don't know" 'Not eligible for coverage / N/A']


****(4, 'Do you know the options for mental health care available under your employer-provided coverage?')****
['Yes' 'No Answer' 'I am not sure' 'No']


****(5, 'Has your employer ever formally discussed mental health (for example, as part of a wellness campaign or other official communication)?')****
['Yes' 'No' "I don't know"]


****(6, 'Does your employer offer resources to learn more about mental health concerns and options for seeking help?')****
['Yes' 'No' "I don't know"]


****(7, 'Is your anonymity protected if you choose to take advantage of mental health or substance abuse treatment resources provided

## First Iteration of Normalizing and Further cleaning
- Columns with one unique answers will be dropped.
- Why or Why not columns need sentiment analysis but for this exercise, will be dropped.
- US terrirotires question will be dropped.
- Genders will be condensed to Male, Female and Other
- questions 44-46 that will be dropped as 47 is representative

In [24]:
df_5 = df_4.drop(one_unique_answer,axis=1)

In [25]:
dropped_columns = ['What US state or territory do you work in?',
                  'What US state or territory do you live in?',
                  'Why or why not?.1',
                  'Why or why not?',
                  'Do you currently have a mental health disorder?',
                  'If yes, what condition(s) have you been diagnosed with?',
                  'If maybe, what condition(s) do you believe you have?']

df_5 = df_5.drop(dropped_columns, axis=1)

In [26]:
def normalize_gender(gender):
    if type(gender) != str:
        raise ValueError("Gender must be a string")
    gender = gender.strip().lower()
    for female in ['female', 'woman','f']:   
        if op.contains(gender,female):
            return 'Female'
    for male in ['male','man','m']:
        if op.contains(gender,male):
            return 'Male'
    return "Other"

In [27]:
df_5.loc[:,"What is your gender?"] = df_5.loc[:,"What is your gender?"].map(normalize_gender)

In [28]:
# doubel checking result
one_unique_answer = []
for i,column in enumerate(df_5.columns):
    if len(df_5[column].unique()) > 1:
        print(f"****{i, column}****")
        print(df_5[column].unique())
        print("\n")
    else:
        one_unique_answer.append(column)
print("The following columns have one unique answer:")
print(*one_unique_answer,sep="\n")

****(0, 'How many employees does your company or organization have?')****
['6-25' 'More than 1000' '26-100' '100-500' '1-5' '500-1000']


****(1, 'Is your employer primarily a tech company/organization?')****
[1. 0.]


****(2, 'Does your employer provide mental health benefits as part of healthcare coverage?')****
['No' 'Yes' "I don't know" 'Not eligible for coverage / N/A']


****(3, 'Do you know the options for mental health care available under your employer-provided coverage?')****
['Yes' 'No Answer' 'I am not sure' 'No']


****(4, 'Has your employer ever formally discussed mental health (for example, as part of a wellness campaign or other official communication)?')****
['Yes' 'No' "I don't know"]


****(5, 'Does your employer offer resources to learn more about mental health concerns and options for seeking help?')****
['Yes' 'No' "I don't know"]


****(6, 'Is your anonymity protected if you choose to take advantage of mental health or substance abuse treatment resources provided

In [29]:
df_5["What country do you live in?"].value_counts()/df_5.shape[0]

What country do you live in?
United States of America    0.674121
United Kingdom              0.104366
Canada                      0.053248
Netherlands                 0.022364
Germany                     0.022364
Australia                   0.021299
Brazil                      0.009585
France                      0.008520
Ireland                     0.008520
Sweden                      0.008520
Switzerland                 0.005325
New Zealand                 0.005325
Finland                     0.004260
Bulgaria                    0.004260
Denmark                     0.004260
India                       0.003195
Spain                       0.003195
South Africa                0.003195
Chile                       0.003195
Russia                      0.003195
Pakistan                    0.003195
Norway                      0.002130
Afghanistan                 0.002130
Italy                       0.002130
Israel                      0.002130
Estonia                     0.002130
Colombia 

In [30]:
df_5['What country do you work in?'].value_counts()/df_5.shape[0]

What country do you work in?
United States of America    0.678381
United Kingdom              0.103301
Canada                      0.052183
Netherlands                 0.022364
Germany                     0.022364
Australia                   0.021299
Brazil                      0.009585
Sweden                      0.009585
France                      0.008520
Ireland                     0.008520
Switzerland                 0.005325
New Zealand                 0.005325
Bulgaria                    0.004260
Finland                     0.004260
Denmark                     0.004260
Russia                      0.003195
Chile                       0.003195
India                       0.003195
South Africa                0.003195
Estonia                     0.002130
Colombia                    0.002130
Spain                       0.002130
Norway                      0.002130
Pakistan                    0.002130
Israel                      0.002130
Austria                     0.001065
Bosnia an

In [31]:
df_5[df_5["What country do you work in?"].isin(["United States of America", "United Kingdom", "Canada", "Netherlands", "Germany", "Australia"])].shape

(845, 45)

In [32]:
df_5[df_5["What country do you live in?"].isin(["United States of America", "United Kingdom", "Canada", "Netherlands", "Germany", "Australia"])].shape

(843, 45)

In [33]:
remaining_obs = df_5[(df_5["What country do you live in?"].isin(["United States of America", "United Kingdom", "Canada", "Netherlands", "Germany", "Australia"]))&
(df_5["What country do you work in?"].isin(["United States of America", "United Kingdom", "Canada", "Netherlands", "Germany", "Australia"]))].shape[0]

total_obs = df_5.shape[0]

print(f"Drop percentage {1 - (remaining_obs/total_obs):.2f}")

Drop percentage 0.10


## Second Iteration of Normalizing and Further cleaning
- Create Dummy variables for questions 43 and 35
- Will only include United States of America, United Kingdom, Canada, Netherlands, Germany, Australia 

In [34]:
df_6 = df_5[(df_5["What country do you live in?"].isin(["United States of America", "United Kingdom", "Canada", "Netherlands", "Germany", "Australia"]))&
            (df_5["What country do you work in?"].isin(["United States of America", "United Kingdom", "Canada", "Netherlands", "Germany", "Australia"]))]
df_6.shape

(842, 45)

In [35]:
work_positions_df = df_6.iloc[:,43].str.get_dummies(sep="|")
work_positions_df.columns = [f"Work_Position_{x}" for x in work_positions_df.columns]
work_positions_df.head()

Work_Position_Back-end Developer  Work_Position_Designer  \
0                                 1                       0   
1                                 1                       0   
2                                 1                       0   
3                                 1                       1   
4                                 1                       0   

   Work_Position_Dev Evangelist/Advocate  Work_Position_DevOps/SysAdmin  \
0                                      0                              0   
1                                      0                              0   
2                                      1                              1   
3                                      0                              1   
4                                      0                              0   

   Work_Position_Executive Leadership  Work_Position_Front-end Developer  \
0                                   0                                  1   
1                                   0                                  0   
2                                   1                                  1   
3                                   0                                  1   
4                                   0                                  0   

   Work_Position_HR  Work_Position_One-person shop  Work_Position_Other  \
0                 0                              0                    0   
1                 0                              0                    0   
2                 0                              0                    0   
3                 0                              0                    0   
4                 0                              0                    0   

   Work_Position_Sales  Work_Position_Supervisor/Team Lead  \
0                    0                                   0   
1                    0                                   0   
2                    0                                   1   
3                    0                                   0   
4                    0                                   0   

   Work_Position_Support  
0                      0  
1                      0  
2                      1  
3                      1  
4                      0

In [36]:
df_7 = pd.concat([df_6,work_positions_df],axis=1).drop("Which of the following best describes your work position?",axis=1)

In [37]:
df_7.head()

How many employees does your company or organization have?  \
0                                               6-25           
1                                               6-25           
2                                               6-25           
3                                     More than 1000           
4                                             26-100           

   Is your employer primarily a tech company/organization?  \
0                                                1.0         
1                                                1.0         
2                                                0.0         
3                                                1.0         
4                                                1.0         

  Does your employer provide mental health benefits as part of healthcare coverage?  \
0                                                 No                                  
1                                                 No                                  
2                                                Yes                                  
3                                                Yes                                  
4                                       I don't know                                  

  Do you know the options for mental health care available under your employer-provided coverage?  \
0                                                Yes                                                
1                                          No Answer                                                
2                                                Yes                                                
3                                      I am not sure                                                
4                                                 No                                                

  Has your employer ever formally discussed mental health (for example, as part of a wellness campaign or other official communication)?  \
0                                                Yes                                                                                       
1                                                 No                                                                                       
2                                                 No                                                                                       
3                                                 No                                                                                       
4                                                 No                                                                                       

  Does your employer offer resources to learn more about mental health concerns and options for seeking help?  \
0                                                Yes                                                            
1                                                 No                                                            
2                                                 No                                                            
3                                                Yes                                                            
4                                                 No                                                            

  Is your anonymity protected if you choose to take advantage of mental health or substance abuse treatment resources provided by your employer?  \
0                                                Yes                                                                                               
1                                       I don't know                                                                                               
2                                                 No                                                                                               
3 

In [38]:
mental_health_df = df_7.iloc[:,35].str.get_dummies(sep="|")
mental_health_df.columns = [f"{x}_Diagnosed_Mental_Health" for x in mental_health_df.columns]
mental_health_df.head()

ADD (w/o Hyperactivity)_Diagnosed_Mental_Health  \
0                                                0   
1                                                0   
2                                                0   
3                                                0   
4                                                0   

   Addictive Disorder_Diagnosed_Mental_Health  \
0                                           0   
1                                           0   
2                                           0   
3                                           0   
4                                           0   

   Anxiety Disorder (Generalized, Social, Phobia, etc)_Diagnosed_Mental_Health  \
0                                                  1                             
1                                                  0                             
2                                                  1                             
3                                                  0                             
4                                                  0                             

   Asperger Syndrome_Diagnosed_Mental_Health  \
0                                          0   
1                                          0   
2                                          0   
3                                          0   
4                                          0   

   Attention Deficit Disorder_Diagnosed_Mental_Health  \
0                                                  0    
1                                                  0    
2                                                  0    
3                                                  0    
4                                                  0    

   Attention Deficit Hyperactivity Disorder_Diagnosed_Mental_Health  \
0                                                  0                  
1                                                  0                  
2                                                  0                  
3                                                  0                  
4                                                  0                  

   Autism_Diagnosed_Mental_Health  \
0                               0   
1                               0   
2                               0   
3                               0   
4                               0   

   Autism (Asperger's)_Diagnosed_Mental_Health  \
0                                            0   
1                                            0   
2                                            0   
3                                            0   
4                                            0   

   Autism - while not a "mental illness", still greatly affects how I handle anxiety_Diagnosed_Mental_Health  \
0                                                  0                                                           
1                                                  0                                                           
2                                                  0                                                           
3                                                  0                                                           
4                                                  0                                                           

   Autism Spectrum Disorder_Diagnosed_Mental_Health  \
0                                                 0   
1                                                 0   
2                                                 0   
3                                                 0   
4                                                 0   

   Dissociative Disorder_Diagnosed_Mental_Health  \
0                                              0   
1                                              0   
2                                              0   
3                                              0   
4                                          

In [39]:
df_8 = pd.concat([df_7,mental_health_df],axis=1).drop(['Have you been diagnosed with a mental health condition by a medical professional?',
                                                       'If so, what condition(s) were you diagnosed with?'],axis=1)

In [40]:
# doubel checking result
for i,column in enumerate(df_8.columns):
    print(f"****{i, column}****")
    print(df_8[column].unique())
    print("\n")


****(0, 'How many employees does your company or organization have?')****
['6-25' 'More than 1000' '26-100' '100-500' '1-5' '500-1000']


****(1, 'Is your employer primarily a tech company/organization?')****
[1. 0.]


****(2, 'Does your employer provide mental health benefits as part of healthcare coverage?')****
['No' 'Yes' "I don't know" 'Not eligible for coverage / N/A']


****(3, 'Do you know the options for mental health care available under your employer-provided coverage?')****
['Yes' 'No Answer' 'I am not sure' 'No']


****(4, 'Has your employer ever formally discussed mental health (for example, as part of a wellness campaign or other official communication)?')****
['Yes' 'No' "I don't know"]


****(5, 'Does your employer offer resources to learn more about mental health concerns and options for seeking help?')****
['Yes' 'No' "I don't know"]


****(6, 'Is your anonymity protected if you choose to take advantage of mental health or substance abuse treatment resources provided

In [41]:
df_8.shape

(842, 79)

In [42]:
# mhc_types = df.iloc[:,48].apply(lambda x:x.split("|")).to_list()

In [43]:
# mhc_types

In [44]:
# mhc_types = list(chain.from_iterable(mhc_types))

In [45]:
# mhc_types

In [46]:
# mhc_db = dict(Counter(mhc_types))

In [47]:
# other = 0
# new_mhc_db = {}
# for key,value in mhc_db.items():
#     if value <= 15:
#         other += value
#     else:
#         new_mhc_db[key] = value    

In [48]:
# print(other)

In [49]:
# new_mhc_db["Other"] = other

In [50]:
# new_mhc_db

In [51]:
# plt.pie(new_mhc_db.values(), labels=new_mhc_db.keys(), autopct='%1.1f%%')
# plt.title('MHC Types')
# plt.show()

In [52]:
# gender_df = df.query("'Mood Disorder (Depression, Bipolar Disorder, etc)' in `If yes, what condition(s) have you been diagnosed with?`")

In [53]:
# gender_df["What is your gender?"].value_counts()

In [54]:
# plt.pie(gender_df["What is your gender?"].value_counts().values, labels=gender_df["What is your gender?"].value_counts().index, autopct='%1.1f%%')
# plt.title('Gender Segmentation of Mood Disorders')
# plt.show()

In [55]:
# df.iloc[:,9].unique()

In [56]:
# pd.get_dummies(df.iloc[:,9],prefix="Question 9")